### ISSUE TITLE
Link sort

### ISSUE THREAD

**Issue by cornflakes** [realm/realm-java#672]
It would be great to have the posibility to sort on a Link.

ie.:
RealmResults<Foo> fooRealmResults = realm.where(Foo.class).("bar.startTimeStamp", start).findAll();
fooRealmResults.sort("bar.startTimeStamp");

**Comment by beeender** [comment]
+1 from http://stackoverflow.com/questions/31435374/realm-findallsorted-by-field-of-a-filed

**Comment by Onejjy** [comment]
:+1: :)

**Comment by draskosaric** [comment]
:+1: :smiley:

**Comment by tgbarker** [comment]
+1

**Comment by WesleyBuck** [comment]
:+1:

**Comment by viniciusam** [comment]
+1

In my case, I wish to retrieve a list of A sorted by A.date with its childs sorted by B.date, where:

```java
Class A {
    Date date;
    RealmList<B> bList;
}
```
```java
Class B {
    Date date;
}
```

Example query:
```java
RealmResults<A> results= realm.where(A.class)
   .findAllSorted(new String[]{"date", "bList.date"}, new Sort[]{Sort.ASCENDING, Sort.ASCENDING});
```

**Comment by nealsanche** [comment]
+1

**Comment by cioraneanu** [comment]
+1 as well. I have a lot of redundant fields that duplicates data in their children for sorting purposes. Updating all these values is a bit of a pain.

**Comment by mmlii** [comment]
+1

**Comment by cmelchior** [comment]
+15 from #709

**Comment by johnryan** [comment]
@cmelchior is there any timeline for this or a recommended workaround? I have a similar case where i have an array of child objects with a date property, thanks.

**Comment by TR4Android** [comment]
:+1: I'm currently in the process of migrating an app from SQLite to Realm and I was kind of expecting that this would work. Since it doesn't I'm currently a bit stuck and need to wait till this is implemented. Any news on this? Is this being worked on? Thanks!

**Comment by kmenager** [comment]
+1

**Comment by zagen0x** [comment]
+1

**Comment by Betobit** [comment]
+1

**Comment by kieranmchall** [comment]
+1 this is really important

**Comment by alexandrahably** [comment]
+1

**Comment by blork** [comment]
+1

**Comment by punkusha** [comment]
+ 1

**Comment by mustafaerturk** [comment]
+1

**Comment by jaypoojara** [comment]
+1

**Comment by beeender** [comment]
Move to P1 since it is supported from core.

**Comment by HeinrichReimer** [comment]
Is there any estimated release date for this feature? Shouldn't be too difficult as it's already part of the core.

**Comment by ZeDzislaw** [comment]
+1

**Comment by nihad92** [comment]
+1

**Comment by heitorpr** [comment]
+1

**Comment by Maverick25** [comment]
+1

**Comment by vitorotero** [comment]
+1

**Comment by ScheNi** [comment]
+1

**Comment by yadavharish** [comment]
+1

**Comment by UndeRus** [comment]
+1

**Comment by JFernandoSE** [comment]
+1

**Comment by fpasteneso** [comment]
+1

**Comment by RJFares** [comment]
+1

This is extremely important to my project, I was able to reduce file size of realm to peanuts, until I noticed I couldn't sort by child.
I have a RecyclerView with icons on the left grouping items by category, and my flow is similar to:
a- Artist
   b- Album
       c- Song Title

**Comment by RJFares** [comment]
Also noticed that I can't return a .distinct("artist.album") :/

**Comment by Zhuinden** [comment]
If you have a link to the `Artist` from the `Album`, then you can search for the album via link queries and then sort directly on the album's property

**Comment by RJFares** [comment]
Unfortunately no I don't, I gave random classes now just as an example, the one I'm currently working on is this:
A- ShoppingItem
B- Food
C- String category.

The shopping list used to be sorted by fetching each shoppingItem, getting the Food object and extracting the food.category, something such as realm.where(ShoppingItem.class).findAllSorted("food.category",Sort.Ascending);
which doesn't work. My workaroound was to put a category also under shoppingItem each time a shoppingItem was created from a food. But then the link is not dynamic, each time the food's category is changed, i have to manually change the category of each shoppingItem related to it.

**Comment by mrtysn** [comment]
+1

**Comment by facos86** [comment]
+1

**Comment by hsandyque** [comment]
+1

**Comment by tmtrademarked** [comment]
Another possible solution here would be to allow passing a comparator function into sort. Then sorting by child fields becomes implementable on the client side. Given that sort appears to be implemented in native code, this might not be practical - but just wanted to suggest it as a possible approach here.

**Comment by Zhuinden** [comment]
@tmtrademarked  https://github.com/realm/realm-java/issues/2313

**Comment by aligorji** [comment]
+1

**Comment by lopezzbelgium** [comment]
+1

**Comment by kirvis250** [comment]
It's a bit disappointing having to wait 2 years for feature that should have been there from start. :)

**Comment by lDuffy** [comment]
+1

**Comment by anninpavel** [comment]
+1

**Comment by chronvas** [comment]
+1

**Comment by adamfaryna** [comment]
+1

**Comment by beeender** [comment]
#3834 is merged to master. will be fixed in `v2.4.0`. For anyone wants to try it now, the `2.4.0-SNAPSHOT` should include this feature.
